### PR TITLE
updated config for planned endpoints and scopes.

### PR DESCRIPTION
**Pull request by grab-a-byte**

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -164,35 +164,66 @@ versioning {
...
+        endpoint = "/individuals/benefits-and-credits/child-tax-credit"
+        fields {
+            D = "applications/awards/childTaxCredit/childCareAmount"
+            E = "applications/awards/childTaxCredit/ctcChildAmount"
+            F = "applications/awards/childTaxCredit/familyAmount"
+            G = "applications/awards/childTaxCredit/babyAmount"
+            H = "applications/awards/childTaxCredit/paidYTD"
+            I = "applications/awards/payments/amount"
+            J = "applications/awards/payments/frequency"
+            K = "applications/awards/payments/startDate"
```
Comment (ajchilts) [inline conf/application.conf:209]: You might be able to explain this better as you have looked into this however; why do we have duplicate paths with different keys in each case? 

I.e. 

K = "applications/awards/payments/startDate"

and

Q = "applications/awards/payments/startDate"

File: conf/application.conf [hunk 2]
```diff
@@ -164,35 +164,66 @@ versioning {
...
-        D = "employer/employerAddress/line2"
-        E = "employer/employerAddress/line3"
-        F = "employer/employerDistrictNumber"
-        G = "employer/employerSchemeReference"
-        H = "employmentStartDate"
-        I = "employmentLeavingDate"
-        J = "employmentPayFrequency"
-        K = "employeeAddress"
-        L = "payrollId"
+    benefitsAndCredits {
```
Comment (ajchilts) [inline conf/application.conf:191]: Also this endpoint config looks different to that in the test config used bu the code for testing the handler pattern. 

```
    endpoints = List(
      EndpointConfig(
        name = mockEndpoint1,
        link = "/a/b/c?matchId=<matchId>{&fromDate,toDate}",
        fields = Map(
          "A" -> "payments",
          "B" -> "employer/employerName=\"foo bar\"",
          "C" -> "employer/employerAddress/line1",
          "D" -> "employer/employerAddress/line2",
          "E" -> "employer/employerAddress/line3",
          "F" -> "employer/employerDistrictNumber",
          "G" -> "employer/employerSchemeReference",
          "H" -> "employmentStartDate",
          "I" -> "employmentLeavingDate",
          "J" -> "employmentPayFrequency",
          "K" -> "employeeAddress",
          "L" -> "payrollId"
        )
      ),
```

I think we will need to align to this name, link, fields 

Name is the string name of the endpoint used by the endpoint to fetch the scopes. 

Link is the full url for HAL 

Fields are the fields. 

@owen-opencast  can help confirm this?

**Review by ajchilts — CHANGES_REQUESTED** [review]

File: conf/application.conf [hunk 3]
```diff
@@ -164,35 +164,66 @@ versioning {
...
+        endpoint = "individuals/benefits-and-credits"
```
Comment (ajchilts) [inline conf/application.conf]: Update to correct HAL link and ensure endpoint keys match those in the controllers `getEndPointScopes` invocation.

**Review by ajchilts — CHANGES_REQUESTED** [review]

File: conf/application.conf [hunk 4]
```diff
@@ -164,35 +164,66 @@ versioning {
...
+      workingTaxCredits {
+        endpoint = "/individuals/benefits-and-credits/working-tax-credit"
+        fields {
+            M = "applications/awards/workingTaxCredit/amount"
+            N = "applications/awards/childTaxCredit/childCareAmount"
+            O = "applications/awards/payments/amount"
+            P = "applications/awards/payments/frequency"
+            Q = "applications/awards/payments/startDate"
+            R = "applications/awards/payments/endDate"
+            S = "applications/awards/workingTaxCredit/piadYTD"
```
Comment (owen-gilfellon) [inline conf/application.conf]: Spelling

File: conf/application.conf [hunk 5]
```diff
@@ -164,35 +164,66 @@ versioning {
...
+            U = "applications/awards/payments/tcType"
+        }
       }
-    }
     hello {
```
Comment (owen-gilfellon) [inline conf/application.conf]: Remove placeholder config?

**Review by owen-gilfellon — CHANGES_REQUESTED** [review]

File: conf/application.conf [hunk 6]
```diff
@@ -164,48 +164,66 @@ versioning {
...
       }
-    }
-    hello {
-      endpoint = "/individuals/hello-world/"
-      fields {
-        A = "one"
-        B = "one/two"
-        C = "two/three/four"
+      child-tax-credits {
+        endpoint = "/individuals/benefits-and-credits/child-tax-credit?matchId=$randomMatchId"
```
Comment (ajchilts) [inline conf/application.conf]: I believe this link should have a startDate and endDate range.

File: conf/application.conf [hunk 7]
```diff
@@ -164,48 +164,66 @@ versioning {
...
+            L = "applications/awards/payments/endDate"
+            T = "applications/awards/payments/tcType"
+        }
       }
-      endpoint = "/individuals/hello-scopes/"
-      fields {
-        A = "one"
-        B = "one/two"
+      working-tax-credits {
+        endpoint = "/individuals/benefits-and-credits/working-tax-credit?matchId=$randomMatchId"
```
Comment (ajchilts) [inline conf/application.conf]: I believe this link should have a startDate and endDate range.

File: conf/application.conf [hunk 8]
```diff
@@ -164,48 +164,66 @@ versioning {
...
-        E = "employer/employerAddress/line3"
-        F = "employer/employerDistrictNumber"
-        G = "employer/employerSchemeReference"
-        H = "employmentStartDate"
-        I = "employmentLeavingDate"
-        J = "employmentPayFrequency"
-        K = "employeeAddress"
-        L = "payrollId"
+    benefitsAndCredits {
+        endpoint = "individuals/benefits-and-credits?matchId=$randomMatchId"
```
Comment (ajchilts) [inline conf/application.conf]: I believe this link should have a startDate and endDate range.

**Review by ajchilts — CHANGES_REQUESTED** [review]

**Review by ajchilts — APPROVED** [review]

File: test/component/uk/gov/hmrc/individualsbenefitsandcreditsapi/controllers/LiveChildTaxCreditControllerSpec.scala [hunk 9]
```diff
@@ -41,6 +49,7 @@ class LiveChildTaxCreditControllerSpec extends BaseSpec {
       Then("The response status should be 500")
       response.code shouldBe INTERNAL_SERVER_ERROR
 
+      println(response.body)
```
Comment (owen-gilfellon) [inline test/component/uk/gov/hmrc/individualsbenefitsandcreditsapi/controllers/LiveChildTaxCreditControllerSpec.scala:52]: Remove println

File: conf/application.conf [hunk 10]
```diff
@@ -164,48 +164,66 @@ versioning {
...
+          }
       }
-    }
-    hello {
-      endpoint = "/individuals/hello-world/"
-      fields {
-        A = "one"
-        B = "one/two"
-        C = "two/three/four"
+      child-tax-credit {
```
Comment (owen-gilfellon) [inline conf/application.conf:199]: benefitsAndCredits is camel case, but hyphens here?

**Pull request merged**